www/base/src/app/builders/builders.tpl.jade

```diff
@@ -37,7 +39,10 @@
             span(ng-show="tags_filter.length > 0")
                 span.label.clickable.label-danger(ng-click="tags_filter = []") x
           th(style="width:20%px;") Workers
-      tr(ng-repeat='builder in builders | filter: isBuilderFiltered | orderBy: "name"')
+      ul(uib-pagination, total-items="builders.length",
+        ng-model="currentPage", class="pagination-sm", boundary-link-numbers="true",
+        max-size=10, items-per-page="page_size")
+      tr(ng-repeat='builder in builders | filter: {name: searchQuery} | filter: isBuilderFiltered | orderBy: "name" | limitTo: page_size :(currentPage-1)*page_size')
```

Comment:
filters are exactly like unix pipe. They are run one by one, and each produce a new array. the last array is passed to ng-repeat. in order to properly implement paging, you will have to use ng-init
```suggestion
      ul(ng-init="filtered_builders = (builders | filter: {name: searchQuery} | filter: isBuilderFiltered)" uib-pagination, total-items="filtered_builders.length", [...])
      tr(ng-repeat='builder in filtered_builders| orderBy: "name" | limitTo: page_size :(currentPage-1)*page_size')
```

Reply:
ng-init didn't do the trick (it was giving many weird errors), so I used filter in total-items. Now it works (doesn't show empty pages)